REGRESSION (258767@main): image on studioneat.com is strangely masked

darinadler opened the pull request:
#### 2e5421f37cc928528af98194c3bc1aa5fb1c571e
<pre>
REGRESSION (258767@main): image on studioneat.com is strangely masked
<a href="https://bugs.webkit.org/show_bug.cgi?id=254063">https://bugs.webkit.org/show_bug.cgi?id=254063</a>
rdar://106464942

Reviewed by Oriol Brufau.

The 258767@main change caused -webkit-mask-box-image to set -webkit-mask-box-image-slices
to &quot;auto&quot; internally instead of to &quot;0 fill&quot;, indirectly, by using an initial value.
This was not detectable by computed style, because the computed style code covered it up
and converted auto to 0. But it caused incorrect rendering.

* LayoutTests/fast/css/mask-box-image-parsing-expected.txt: Added.
* LayoutTests/fast/css/mask-box-image-parsing.html: Added.
* LayoutTests/fast/css/mask-box-image-slices-expected.html: Added.
* LayoutTests/fast/css/mask-box-image-slices.html: Added.

* Source/WebCore/css/ComputedStyleExtractor.cpp:
(WebCore::valueForImageSliceSide): Removed isAuto case. No auto value will eveer get here.

* Source/WebCore/css/parser/CSSPropertyParser.cpp:
(WebCore::initialValueForLonghand): Removed incorrect default value for
CSSPropertyWebkitMaskBoxImageSlice.
(WebCore::isInitialValueForLonghand): Added cases for all 4 of the -webkit-mask-box-image
longhands, since the parser expands them out into objects. This is the same as what we
already had here for border-image longhands.
(WebCore::initialValueTextForLonghand): Added a special case for
CSSPropertyWebkitMaskBoxImageSlice.
(WebCore::initialValueIDForLonghand): Ditto.

* Source/WebCore/style/StyleBuilderCustom.h:
(WebCore::Style::ApplyPropertyBorderImageModifier::applyInitialValue): Use a box of
fixed 0 for -webkit-mask-box-image-slices instead of auto. The auto here was incorrect
and is what led to the strange results above.

Canonical link: <a href="https://commits.webkit.org/261808@main">https://commits.webkit.org/261808@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d5f494954278079ed6ee035b40452c1e85d51218

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/112908 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/22067 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/90/builds/1584 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/4685 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/121406 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/117010 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/76/builds/23638 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/13228 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/86/builds/5875 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/118695 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/76/builds/23638 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/100643 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/105998 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/76/builds/23638 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/88/builds/1181 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/46411 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/14362 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/89/builds/1221 "Passed tests") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/15066 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/10560 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/20378 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/53213 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/8244 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/16917 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

Loirooriol reviewed (approved):
Yeah, this was broken, I also reported it in https://bugs.webkit.org/show_bug.cgi?id=248346

This is doing the 1st option that I mentioned there, which seems more reasonable and matches https://drafts.fxtf.org/css-masking-1/#the-mask-border-slice

darinadler committed:
REGRESSION (258767@main): image on studioneat.com is strangely masked
https://bugs.webkit.org/show_bug.cgi?id=254063
rdar://106464942

Reviewed by Oriol Brufau.

The 258767@main change caused -webkit-mask-box-image to set -webkit-mask-box-image-slices
to "auto" internally instead of to "0 fill", indirectly, by using an initial value.
This was not detectable by computed style, because the computed style code covered it up
and converted auto to 0. But it caused incorrect rendering.

* LayoutTests/fast/css/mask-box-image-parsing-expected.txt: Added.
* LayoutTests/fast/css/mask-box-image-parsing.html: Added.
* LayoutTests/fast/css/mask-box-image-slices-expected.html: Added.
* LayoutTests/fast/css/mask-box-image-slices.html: Added.

* Source/WebCore/css/ComputedStyleExtractor.cpp:
(WebCore::valueForImageSliceSide): Removed isAuto case. No auto value will eveer get here.

* Source/WebCore/css/parser/CSSPropertyParser.cpp:
(WebCore::initialValueForLonghand): Removed incorrect default value for
CSSPropertyWebkitMaskBoxImageSlice.
(WebCore::isInitialValueForLonghand): Added cases for all 4 of the -webkit-mask-box-image
longhands, since the parser expands them out into objects. This is the same as what we
already had here for border-image longhands.
(WebCore::initialValueTextForLonghand): Added a special case for
CSSPropertyWebkitMaskBoxImageSlice.
(WebCore::initialValueIDForLonghand): Ditto.

* Source/WebCore/style/StyleBuilderCustom.h:
(WebCore::Style::ApplyPropertyBorderImageModifier::applyInitialValue): Use a box of
fixed 0 for -webkit-mask-box-image-slices instead of auto. The auto here was incorrect
and is what led to the strange results above.

Canonical link: https://commits.webkit.org/261808@main
